app/src/main/java/com/weatherxm/data/ApiModels.kt

```diff
@@ -66,7 +82,17 @@ data class PublicDevice(
             id = id,
             name = name,
             cellIndex = cellIndex,
-            profile = profile,
+            bundleName = try {
+                BundleName.valueOf(bundle?.name ?: String.empty())
+            } catch (e: IllegalArgumentException) {
+                Timber.e("Wrong Bundle Name: ${bundle?.name} for Device $name")
+                null
+            },
+            bundleTitle = bundle?.title,
+            connectivity = bundle?.connectivity,
+            wsModel = bundle?.wsModel,
+            gwModel = bundle?.gwModel,
+            hwClass = bundle?.hwClass,
```

Comment:
Mostly due to the fact that we use the `BundleName enum class` in the `UIDevice` (the `try...catch` here parsing it) whereas in the `Device` model in the API we use `String?` type so that we will be able to parse any model and not getting an error/exception. But we use the `enum class BundleName` in the `UIDevice` in order to be able to perform equality checks throughout our UI.

So that's the main reason I do not pass the object and pass only the ones we need, the way we need them (e.g. `hwClass` could be removed from here as it's not used anywhere, I added it just for completeness.